pkg/apis/capabilities/v1beta1/product_types.go

```diff
@@ -0,0 +1,1095 @@
+	// The operator will retry.
+	ProductFailedConditionType common.ConditionType = "Failed"
+)
+
+var (
+	//
+	productSystemNameRegexp = regexp.MustCompile("[^a-zA-Z0-9]+")
+)
+
+// MetricMethodRefSpec defines method or metric reference
+// Metric or method can optionally belong to used backends
+type MetricMethodRefSpec struct {
+	// SystemName identifies uniquely the metric or methods
+	SystemName string `json:"systemName"`
+
+	// BackendSystemName identifies uniquely the backend
+	// Backend reference must be used by the product
+	// +optional
+	BackendSystemName *string `json:"backend,omitempty"`
+}
+
+func (m *MetricMethodRefSpec) String() string {
+	backendPrefix := ""
+	if m.BackendSystemName != nil {
+		backendPrefix = fmt.Sprintf("%s.", *m.BackendSystemName)
+	}
+	return fmt.Sprintf("%s%s", backendPrefix, m.SystemName)
+}
+
+// LimitSpec defines the maximum value a metric can take on a contract before the user is no longer authorized to use resources.
+// Once a limit has been passed in a given period, reject messages will be issued if the service is accessed under this contract.
+type LimitSpec struct {
+	// Limit Period
+	// +kubebuilder:validation:Enum=eternity;year;month;week;day;hour;minute
+	Period string `json:"period"`
+
+	// Limit Value
+	Value int `json:"value"`
+
+	// Metric or Method Reference
+	MetricMethodRef MetricMethodRefSpec `json:"metricMethodRef"`
+}
+
+// PricingRuleSpec defines the cost of each operation performed on an API.
+// Multiple pricing rules on the same metric divide up the ranges of when a pricing rule applies.
+type PricingRuleSpec struct {
+	// Range From
+	From int `json:"from"`
+
+	// Range To
+	To int `json:"to"`
+
+	// Metric or Method Reference
+	MetricMethodRef MetricMethodRefSpec `json:"metricMethodRef"`
+
+	// Price per unit (USD)
+	// +kubebuilder:validation:Pattern=`^\d+.?\d{2}$`
+	PricePerUnit string `json:"pricePerUnit"`
+}
+
+// ApplicationPlanSpec defines the desired state of Product's Application Plan
+type ApplicationPlanSpec struct {
+	// +optional
+	Name *string `json:"name,omitempty"`
+
+	// Set whether or not applications can be created on demand
+	// or if approval is required from you before they are activated.
+	// +optional
+	AppsRequireApproval *bool `json:"appsRequireApproval,omitempty"`
+
+	// Trial Period (days)
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	TrialPeriod *int `json:"trialPeriod,omitempty"`
+
+	// Setup fee (USD)
+	// +kubebuilder:validation:Pattern=`^\d+.?\d{2}$`
+	// +optional
+	SetupFee *string `json:"setupFee,omitempty"`
+
+	// Cost per Month (USD)
+	// +kubebuilder:validation:Pattern=`^\d+.?\d{2}$`
+	// +optional
+	CostMonth *string `json:"costMonth,omitempty"`
+
+	// Pricing Rules
+	// +optional
+	PricingRules []PricingRuleSpec `json:"pricingRules,omitempty"`
+
+	// Limits
+	// +optional
+	Limits []LimitSpec `json:"limits,omitempty"`
+
+	// TODO Features
+}
+
+// Methodpec defines the desired state of Product's Method
+type Methodpec struct {
```

Comment:
Typo in the struct name (and comment). I guess it should be `MethodSpec`

Reply:
:+1: